meson.build

```diff
@@ -141,7 +149,7 @@ if host_os == 'netbsd'
     libsearch_dirs += '/usr/pkg/lib'
 endif
 
-if host_os == 'omnios'
+if uname.found() and run_command(uname, '-n', check: false).stdout().strip().contains('omnios')
```

Comment:
This is quite the mouthful! Did you figure out why the `host_machine.system()` check didn't work for omnios earlier in the script?